homeassistant/components/bluetooth/manager.py

```diff
@@ -230,3 +249,27 @@ def async_register_scanner(
 
         unregister = super().async_register_scanner(scanner, connection_slots)
         return partial(self._async_unregister_scanner, scanner, unregister)
+
+    @hass_callback
+    def _handle_config_entry_changed(
+        self,
+        change: config_entries.ConfigEntryChange,
+        entry: config_entries.ConfigEntry,
+    ) -> None:
+        """Handle config entry changes."""
+        if (
+            change != config_entries.ConfigEntryChange.REMOVED
+            or entry.source != config_entries.SOURCE_IGNORE
+            or not (discovery_keys := entry.discovery_keys)
+        ):
+            return
+        for discovery_key in discovery_keys:
+            if (
+                discovery_key.domain != DOMAIN
+                or discovery_key.version != 1
+                or not isinstance(discovery_key.key, str)
+            ):
```

Comment:
It seems discovery keys might be better as a `dict` by `DOMAIN`

Reply:
Than we could connect the dispatcher to only get updates for the domain

Reply:
Yes, this is a good idea. I'll implement it in a separate PR.